Description: Fix use after free after *solve
 This only occurs if a leg between the two exact same stations appears right
 before and right after the *solve, which is unlikely in real data, but the
 testsuite has an instance of this.
Author: Olly Betts <olly@survex.com>
Origin: upstream
Forwarded: yes
Last-Update: 2015-06-26

--- survex-1.2.20.orig/src/netskel.c
+++ survex-1.2.20/src/netskel.c
@@ -86,6 +86,8 @@ solve_network(void /*node *stnlist*/)
    static int first_solve = 1;
    node *stn;
 
+   clear_last_leg();
+
    if (stnlist == NULL) {
       if (first_solve) fatalerror(/*No survey data*/43);
       /* We've had a *solve followed by another *solve (or the implicit
